nix/sources.json

```diff
@@ -0,0 +1,56 @@
+{
+    "mach-nix": {
+        "branch": "merged",
+        "description": "Create highly reproducible python environments",
+        "homepage": "",
+        "owner": "PrivateStorageio",
+        "repo": "mach-nix",
+        "rev": "0872dd81afe9c4a6552604f7d21fe7f2baddf454",
+        "sha256": "0hsbm6rmjjjzxdciirmcxyvrrlz19cbhprd2hfksrv6nnl4c3mc3",
+        "type": "tarball",
+        "url": "https://github.com/PrivateStorageio/mach-nix/archive/0872dd81afe9c4a6552604f7d21fe7f2baddf454.tar.gz",
+        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
+    },
+    "niv": {
+        "branch": "master",
+        "description": "Easy dependency management for Nix projects",
+        "homepage": "https://github.com/nmattia/niv",
+        "owner": "nmattia",
+        "repo": "niv",
+        "rev": "e0ca65c81a2d7a4d82a189f1e23a48d59ad42070",
+        "sha256": "1pq9nh1d8nn3xvbdny8fafzw87mj7gsmp6pxkdl65w2g18rmcmzx",
+        "type": "tarball",
+        "url": "https://github.com/nmattia/niv/archive/e0ca65c81a2d7a4d82a189f1e23a48d59ad42070.tar.gz",
+        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
+    },
+    "pypi-deps-db": {
+        "branch": "master",
+        "description": "Probably the most complete python dependency database",
+        "homepage": "",
+        "owner": "DavHau",
+        "repo": "pypi-deps-db",
+        "rev": "96d01556b4597c022647acbf8c3b58d2a99bc963",
+        "sha256": "0s6ll2hi40gj6mp2zdg7w3dq17g381gnfkm390mqgp574lmbq6yw",
+        "type": "tarball",
+        "url": "https://github.com/DavHau/pypi-deps-db/archive/96d01556b4597c022647acbf8c3b58d2a99bc963.tar.gz",
+        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
+    },
+    "release2015": {
+        "sha256": "112drvixj81vscj8cncmks311rk2ik5gydpd03d3r0yc939zjskg",
+        "type": "tarball",
+        "url": "https://releases.nixos.org/nixos/21.05/nixos-21.05.3740.ce7a1190a0f/nixexprs.tar.xz",
+        "url_template": "https://releases.nixos.org/nixos/21.05/nixos-21.05.3740.ce7a1190a0f/nixexprs.tar.xz"
+    },
+    "tahoe-lafs": {
+        "branch": "pull/1129/head",
```

Comment:
What is the "1.16 release branch"? There is `3754.release-1.16.0-rc1`, which I assume a 1.16.1 would be *based* on, but I'd've guess there'd be a new branch for it.

Reply:
Yeah, we can use the tag here (which is what I've done), though obviously that will need to be update manually to get a new version (though you can pass `--branch <whatever>` to `niv update` to do that).

> Or does Tahoe just need to change its release-related workflow in order to let Niv properly manage it?

What does "properly" even mean here? :upside_down_face: If we wanted `niv update tahoe-lafs` to automatically get us the most recent release (either of a particular minor version, or overall), then yes, tahoe would need a release workflow that gave that thing that we want to track a stable name that niv could then resolve to a commit.

I think, though, that ZKAPAuthorizer in particular depends too much on tahoe-lafs internals that aren't covered by any stability guarantee for us to benefit from tahoe doing anything differently.

It occurs to me that this could reasonably `master` here, so that `niv update tahoe-lafs` will update to the latest commit, even though there has been no point at which running that command would have got us this commit.